src/main/java/org/apache/maven/shared/transfer/dependencies/collect/internal/Maven30DependencyCollector.java

```diff
@@ -25,7 +25,9 @@
 import org.apache.maven.RepositoryUtils;
 import org.apache.maven.artifact.handler.ArtifactHandler;
 import org.apache.maven.artifact.handler.manager.ArtifactHandlerManager;
+import org.apache.maven.artifact.repository.ArtifactRepository;
 import org.apache.maven.model.Model;
+import org.apache.maven.project.MavenProject;

```

Comment:
This doesn't look right. Only the `org.apache.maven.shared.transfer.project` is allowed to use classes from Maven-Core.
Some background: Originally there was no dependency on Maven Core, this library was a pure bridge for both Aether flavors. Karl Heinz added the the functionality for MavenProject since the related code was at least used by 2 plugins.
I still have my doubts about it, but I don't have a better solution.
However, that shouldn't open the door to allow the usage of Maven Core in other packages. I haven't looked closely to what's happening here (the PR is HUGE), but this is a codesmell to me.

Reply:
Seems like an unnecessary design constraint. The public API (DependencyCollector) already depends on maven-core (takes ProjectBuildingRequest as a parameter) so why restricting non public APIs to not depend on maven-core? Anyway are you saying the solution is to move the code to a package inside org.apache.maven.shared.transfer.project?

Reply:
Unfortunately I have to agree with Robert, cause the PR huge...and yes I'm not sure if it's the right way to go with the dep to Maven Core...currently there is no other way but I also would restrict it otherwise we got more coupling as we already have. It might be a better solution to make a separate component which handles things in the PR...which uses m-a-t ?...I need to reconsider this more in detail...